src/basic/virt.h

```diff
@@ -66,6 +66,7 @@ static inline bool VIRTUALIZATION_IS_CONTAINER(int x) {
 int detect_vm(void);
 int detect_container(void);
 int detect_virtualization(void);
+int detect_private_users(void);
```

Comment:
in order to make clear that the userns stuff is orthogonal to the virt/container detection I'd probably call this detection function "running_in_userns()", i.e. make it more like the existing `running_in_chroot()` call than the existing `running_in_container()` call...